system/gsettings.py

```diff
@@ -0,0 +1,225 @@
+
+        # Determine the session pid.
+        self.session_pid = -1
+        self._cleanup = False
+
+        for procfile in os.listdir('/proc'):
+            if procfile.isdigit():
+                if proc_name(procfile) in SESSION_MANAGERS:
+                    self.session_pid = procfile
+
+        osuser = pwd.getpwuid(os.getuid()).pw_name
+        if self.session_pid <> -1:
+            sessionuser = proc_owner(self.session_pid)
+        else:
+            sessionuser = osuser
+
+        # Setup the environment and connect to the bus.
+        if self.session_pid <> -1 and osuser == sessionuser:
+            os.environ[ENV_DBUS] = bus_address(self.session_pid)
+        else:
+            self.session_pid = new_session()
+            self._cleanup = True
+
+        self.bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
+
+    def get_value(self, schema, key):
+        ''' Read the value of the specified key '''
+        if ':' in schema:
+            path = schema.split(':')[1]
+            schema = schema.split(':')[0]
+        else:
+            path = None
+        self.settings = Gio.Settings(schema=schema, path=path)
+        return self.settings.get_value(key)
+
+    def set_value(self, schema, key, value):
+        ''' Write a value to the specified key '''
+        if ':' in schema:
+            path = schema.split(':')[1]
+            schema = schema.split(':')[0]
+        else:
+            path = None
+        self.settings = Gio.Settings(schema=schema, path=path)
+        self.settings.set_value(key, value)
+
+    def __del__(self):
+        ''' Class cleanup '''
+        if self._cleanup:
+           # Stop the dbus daemon if one was started.
+           time.sleep(1)
+           os.kill(int(self.session_pid), signal.SIGTERM)
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            settings = dict(required=True, type='list')
+        ),
+        supports_check_mode = True,
+    )
+
+    has_changed = False
+    if not HAS_GIO:
+        module.fail_json(rc=1, msg='The python PyGObject module is a required dependency.')
+
+    gs = Gsettings()
+    gsettings = module.params['settings']
+    for gsetting in gsettings:
+        schema = gsetting['schema']
+        key = gsetting['key']
+        value = gsetting['value']
+        old_setting = gs.get_value(schema, key)
+        setting_type = old_setting.get_type_string()
+
+        old_value = old_setting.unpack()
+        if old_value != value:
+            has_changed = True
+            if module.check_mode:
+                change_msgs.append('{0}/{1} -> current: {2} ; proposed new: {3}'.format(schema, key, old_value, value))
+            else:
+                new_value = GLib.Variant(setting_type, value)
+                gs.set_value(schema, key, new_value)
+                change_msgs.append('{0}/{1} -> old: {2} ; new: {3}'.format(schema, key, old_value, value))
+    gs = None
```

Comment:
This is not a very obvious way to trigger the cleanup.  Can you define and call a `.cleanup()` method instead?